### PR TITLE
Add new guide about overwriting transactions

### LINKED ISSUE #1238: New guide: How to override / edit a transaction in the pool, which hasn't been included in a block yet
### Description

- Users can opt to invalidate a previously sent transaction by resending a new transaction with the same nonce and a higher fee.
- Create a guide on how to override/edit a transaction in the pool, which hasn't been included in a block yet.
- Add a CLI usage example for this guide.

### Motivation
To increase awareness of this interesting feature.

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #1238

### CODE REVIEW

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+* [x] How to overwrite a transaction in the transaction pool
+* [x] How to increase the priority of a transaction in the transaction pool
+====
+
+== Intro
+
+When a transaction is sent to a node, it is first included in the xref:{url_network_txpool}[transaction pool].
+Transactions in the transaction pool wait to be included in a block on the blockchain.
+
+It may be the case, that the transaction sender intends to update the transaction before it is included in a block.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
imho, I would just slightly reword the following two sentences on lines 20 and 21 as follows:
It is possible that the sender of the transaction has the intention of modifying the transaction before it gets included in a block.
```

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+For example, to update specific parameters of the transaction, to replace the transaction entirely, or to speed up the inclusion of the transaction into a block by providing a higher fee.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]: This could involve updating certain parameters of the transaction, completely replacing the transaction, or expediting its inclusion into a block by offering a higher fee.

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+This can be achieved by resending the updated transaction with a higher fee while using the same nonce, as explained in the following steps.
+
+.This only works for transactions in the transaction pool
+IMPORTANT: Overwriting a transaction is only possible for transactions, which haven't been included in a block, yet.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
IMPORTANT: Overwriting a transaction is only possible for transactions which have not yet been included in a block.
```

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+When the updated transaction is accepted by the node, it will be added to the transaction pool.
+If the transaction pool already contains the original transaction with the same nonce with the same sender account, validators can choose which transaction to include into the next block.
+Generally, a validator will always choose to include the transactions with the highest transaction fees, to maximize their rewards for generating new blocks.
+That is why, if two transactions from the same sender with the same nonce exist in the pool, the transaction with the higher fee will be picked preferably to be included into a block.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
Hence, if two transactions from the same sender with the same nonce exist in the pool, the transaction with the higher fee will be picked preferably to be included into a block.
```

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+As soon as the updated transaction is included in a block, it will invalidate the remaining original transaction, because of the outdated nonce.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]: imho, I would just slightly reword this sentence:
Once the updated transaction is incorporated into a block, the remaining original transaction will be invalidated due to its outdated nonce.

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+}
+----
+
+To speed up the inclusion of our transaction into the blockchain, we intend to recreate the exact same transaction with a higher fee.
+
+== Recreating the transaction
+
+Create a new transaction with the same nonce and a higher fee.
+
+A slightly higher fee than specified in the original transaction suffices so that validators will prefer to include the new transaction instead of the old with a lower fee.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
A slightly higher fee than specified in the original transaction will suffice, so that validators will prefer to include the new transaction instead of the old one with a lower fee.
```

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+NOTE: In case the network is currently very busy, it can be useful to adjust the fee even higher, so it is included faster into a block than the other transactions in the transaction pool.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
NOTE: If the network is currently experiencing high congestion, increasing the fee even further can prove beneficial, ensuring its faster inclusion into a block compared to other transactions in the transaction pool.
```

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+If the transaction pool already contains the original transaction with the same nonce with the same sender account, validators can choose which transaction to include into the next block.
+Generally, a validator will always choose to include the transactions with the highest transaction fees, to maximize their rewards for generating new blocks.
+That is why, if two transactions from the same sender with the same nonce exist in the pool, the transaction with the higher fee will be picked preferably to be included into a block.
+
+As soon as the updated transaction is included in a block, it will invalidate the remaining original transaction, because of the outdated nonce.
+As a result, the transaction will be removed from the transaction pool.
+
+== Example: Original transaction
+
+Let's assume we have sent the following transaction to the node, but it is currently pending in the transaction pool, because of too many other transactions waiting in the pool with higher fees:
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
Let's assume we have sent the following transaction to the node, but it is currently pending in the transaction pool as there are already too many other transactions waiting in the pool with higher fees:
```

**Review by chris529 — COMMENTED** [review]
Looks great! I just left a few minor suggestions 👍

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc (hunk 1 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
It is possible that the sender of the transaction has the intention of modifying/updating the transaction before it gets included in a block.
```

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc (hunk 2 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
This could involve updating certain parameters of the transaction, completely replacing the transaction, or expediting its inclusion into a block by offering a higher fee.
```

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc (hunk 5 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
Once the updated transaction is incorporated into a block, the remaining original transaction will be invalidated due to its outdated nonce.
```

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+It is possible that the sender of the transaction has the intention of modifying/updating the transaction before it gets included in a block.
+This could involve updating certain parameters of the transaction, completely replacing the transaction, or expediting its inclusion into a block by offering a higher fee.
+
+This can be achieved by resending the updated transaction with a higher fee while using the same nonce, as explained in the following steps.
+
+.This only works for transactions in the transaction pool
+IMPORTANT: Overwriting a transaction is only possible for transactions which have not yet been included in a block.
+
+When the updated transaction is accepted by the node, it will be added to the transaction pool.
+If the transaction pool already contains the original transaction with the same nonce with the same sender account, validators can choose which transaction to include into the next block.
```
Comment (chris529) [inline docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]: imho, I would just suggest rewording this sentence as follows:
If the transaction pool already includes the original transaction with the same nonce and sender account, validators have the option to select which transaction to include in the upcoming block.

**Review by chris529 — APPROVED** [review]
LGTM! 👍 I just made 1 minor suggestion..

**Review by has5aan — APPROVED** [review]
🆗; besides, the open conversation around line 29.

File: docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc (hunk 9 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/integrate-blockchain/overwriting-transactions.adoc]:
```suggestion
If the transaction pool already includes the original transaction with the same nonce and sender account, validators have the option to select which transaction to include in the upcoming block.
```
